demo/admin/src/products/future/generated/ProductForm.tsx

```diff
@@ -63,9 +63,10 @@ type FormValues = ProductFormDetailsFragment & {
 
 interface FormProps {
     id?: string;
+    showAvailableSince?: boolean;
 }
 
-export function ProductForm({ id }: FormProps): React.ReactElement {
+export function ProductForm({ id, showAvailableSince }: FormProps): React.ReactElement {
```

Comment:
> generated api: maybe change generated api from show... to hide... to show by default

that's a bit tricky... probably I would stick with show... because it's the more defensive approach. E.g. it's some admin-feature it could be a problem if it's shown by accident. Maybe the prop should even not be optional.
hide... on the other hand is more convenient if the optionalRender-fields are shown most of the time.

what do you think? change to hide..., keep show... with optional, or show... (or hide...) with required prop?

Reply:
If the default value is show=false, show is Ok. Else it should be hide=false.

> defensive approach

security wise a defense approach is good. But this does not count for client side code!

Reply:
who defines what the default-value is? can we generally define default: show=false or do you think this should be configurable in config?

Reply:
I dont think making it configurable is neccesary for now. Use the default that fits your project best

Reply:
I'd prefer visible by default and hide via prop, like MUI does for `disableTypography` etc. But that's not a _very_ good argument.